frontend/app/components/api/api-v3/hal-resources/hal-resource.service.ts

```diff
@@ -340,8 +340,9 @@ function initializeResource(halResource:HalResource) {
   function setter(val:HalResource, linkName:string) {
     if (!val) {
       halResource.$source._links[linkName] = {href: null};
-    }
-    else if (val.$link) {
+    } else if (_.isArray(val)) {
```

Comment:
@ulferts This was causing `$source` not being properly updated when writing an array link. Its dubious why this never failed anywhere else.